### PR TITLE
Add 1.26 release notes

### PR DESCRIPTION
**Pull request by codyjlandstrom**

### CODE REVIEW

File: src/content/release-notes.mdx [hunk 1]
```diff
@@ -5,6 +5,38 @@ sidebar_label: Release notes
...
+- Helm Release Name Limit: We are limiting the length of Helm release names and allowing to override the default backend service name to make it shorter if needed. The defaultBackend.nameOverride allows to change the full name of the default backend service. In case you want to change it as part of an upgrade, it might have side effects in the installation, so please, follow this guide
+- Deploying Private Repositories now uses the Okteto backend as the SSH agent instead of mounting the local SSH agent into BuildKit. This change provides feature parity between remote and local deploys but may affect cases where repositories are cloned as part of a command defined in your Okteto Manifest.
+- Buildkit Persistence is now enabled by default. The default disk size is set to 100Gi and cache size is now specified as a ratio of the total disk size (default: 0.9). For example, with a 100Gi disk, 90% will be allocated to cache. If you had previously set `buildkit.persistence.cache setting` please adjust the new ratio accordingly, as that setting will not apply any longer
+
+### New Features
```
Comment (pchico83) [inline src/content/release-notes.mdx:21]: We should document the changes to configure "Deploy on Remote" from the Admin UI

File: src/content/release-notes.mdx [hunk 2]
```diff
@@ -5,6 +5,38 @@ sidebar_label: Release notes
...
+- **Introducing the Okteto Resource Manager**: a new feature that automatically optimizes CPU and memory requests for your environments. By analyzing real-time resource utilization, the Resource Manager dynamically adjusts resource requests to ensure efficient usage, prevent node overload, and improve overall cluster performance. This feature simplifies resource management, reduces manual adjustments, and enhances application stability, especially in larger clusters
+- Added Okteto Garbage Collector settings to the Admin Dashboard: Admins of Okteto can now manage sleep and delete periods from the Dashboard
+- Added additional feature and documentation for [running Buildkit at scale](self-hosted/manage/buildkit-high-performance.mdx)  <!-- 8727 -->
+
+### Improvements
+
+- Create docker secret if one doesn't already exist in the controller manager  <!-- 8566 -->
+- Buildkit cache size is now automatically configured based on its PVC volume size  <!-- 8723 -->
+- Removed buildkit persistency as an installation step now that it defaults to true  <!-- 8771 -->
+- We now allow the Okteto frontend and Buildkit manifest with flags and properties to run rootless   <!-- 8774, 8800 -->
```
Comment (pchico83) [inline src/content/release-notes.mdx:32]: - Frontend always run as rootless and it's not configurable
- Buildkit, if rootless is enabled, doesn't use `privilege`

**Comment by codyjlandstrom** [comment]
Making "Ready to Review" to get a preview link

File: src/content/release-notes.mdx (hunk 1 above)
Comment (codyjlandstrom) [reply src/content/release-notes.mdx:21]: @ifbyol Will this be done in time?

Comment (ifbyol) [reply src/content/release-notes.mdx:21]: Yes, this is already done and included in the upcoming release

File: src/content/release-notes.mdx [hunk 3]
```diff
@@ -5,6 +5,38 @@ sidebar_label: Release notes
...
+- Added additional feature and documentation for [running Buildkit at scale](self-hosted/manage/buildkit-high-performance.mdx)  <!-- 8727 -->
+
+### Improvements
+
+- Okteto will now automatically create a Docker secret if one doesn't already exist in the Controller Manager  <!-- 8566 -->
+- Buildkit cache size is now automatically configured based on its PVC volume size  <!-- 8723 -->
+- Removed Buildkit persistency as an installation step now that it defaults to true  <!-- 8771 -->
+- The Okteto frontend now runs rootless by default, while Buildkit operates without privileges when rootless mode is enabled  <!-- 8774, 8800 -->
+
+### Bug Fixes
```
Comment (ifbyol) [inline src/content/release-notes.mdx:34]: We included another fix. Build logs were not being displayed in the UI when a dev environment was deployed from the UI or a preview environment was deployed. The UI seemed frozen until the build finished, but the logs could not be seen.

We have fixed it and now, build logs appear again while deploying dev environments and preview environments

File: src/content/release-notes.mdx [hunk 4]
```diff
@@ -5,6 +5,38 @@ sidebar_label: Release notes
...
+- **Buildkit Persistence Enabled**: Buildkit persistence is now enabled by default, with a 100Gi disk and cache set to 90% of the disk size. If you previously used buildkit.persistence.cache, adjust to the new ratio, as this setting is no longer applicable.
+
+### New Features
+
+- **Introducing the [Okteto Resource Manager](admin/resource-manager.mdx)**: a new feature that automatically optimizes CPU and memory requests for your environments. By analyzing real-time resource utilization, the Resource Manager dynamically adjusts resource requests to ensure efficient usage, prevent node overload, and improve overall cluster performance. This feature simplifies resource management, reduces manual adjustments, and enhances application stability, especially in larger clusters
+- Added Okteto [Garbage Collector settings to the Admin Dashboard](admin/cleanup.mdx): Admins of Okteto can now manage sleep and delete periods from the Dashboard
+- Added additional feature and documentation for [running Buildkit at scale](self-hosted/manage/buildkit-high-performance.mdx)  <!-- 8727 -->
+
+### Improvements
```
Comment (ifbyol) [inline src/content/release-notes.mdx:27]: As part of the CLI changes, we have added the ability of setting `context` field in the Okteto Manifest for `deploy` and `destroy` section to be used as the working directory of the commands specified on those sections. Something similar to the same field in the `test` section: https://www.okteto.com/docs/1.26/reference/okteto-manifest/#context-string-optional

We should probably add documentation about that field, I think it wasn't added.

Another improvement we have added in the CLI is the retry on build operations if BuildKit is not available. CLI will automatically retry builds if there is a transient error connecting with BuildKit. We have added the following env vars to control this behavior: `OKTETO_BUILDKIT_MAX_RETRIES_FOR_TRANSIENT_ERRORS`, `OKTETO_BUILDKIT_WAIT_TIMEOUT` and `OKTETO_BUILDKIT_RETRY_INTERVAL` . We can add a link to this page: https://www.okteto.com/docs/1.26/reference/feature-flags/

File: src/content/release-notes.mdx (hunk 1 above)
Comment (codyjlandstrom) [reply src/content/release-notes.mdx:21]: Ok thanks. I just moved the ticket status to done and will update here

File: src/content/release-notes.mdx [hunk 5]
```diff
@@ -5,6 +5,38 @@ sidebar_label: Release notes
...
+- **Private Repository Deploys**: Deploying private repositories now uses the Okteto backend as the SSH agent, rather than mounting the local SSH agent. This brings feature parity between remote and local deploys but may affect repositories cloned within Okteto Manifest commands.
+- **Buildkit Persistence Enabled**: Buildkit persistence is now enabled by default, with a 100Gi disk and cache set to 90% of the disk size. If you previously used buildkit.persistence.cache, adjust to the new ratio, as this setting is no longer applicable.
+
+### New Features
+
+- **Introducing the [Okteto Resource Manager](admin/resource-manager.mdx)**: a new feature that automatically optimizes CPU and memory requests for your environments. By analyzing real-time resource utilization, the Resource Manager dynamically adjusts resource requests to ensure efficient usage, prevent node overload, and improve overall cluster performance. This feature simplifies resource management, reduces manual adjustments, and enhances application stability, especially in larger clusters
```
[the remark this answers is not in the document]
Comment (codyjlandstrom) [reply src/content/release-notes.mdx:23]: Good call out!

File: src/content/release-notes.mdx [hunk 6]
```diff
@@ -5,6 +5,38 @@ sidebar_label: Release notes
...
+- Added Okteto [Garbage Collector settings to the Admin Dashboard](admin/cleanup.mdx): Admins of Okteto can now manage sleep and delete periods from the Dashboard
```
[the remark this answers is not in the document]
Comment (codyjlandstrom) [reply src/content/release-notes.mdx:24]: We left those settings in the helm config. Should they be removed?

**Comment by codyjlandstrom** [comment]
@ifbyol I think the latest changes address your feedback. Please take a look and let me know what you think. Also, is there anything that we're not covering for CLI 3.1.0?

File: src/content/release-notes.mdx [hunk 7]
```diff
@@ -5,6 +5,43 @@ sidebar_label: Release notes
...
+- Added additional feature and documentation for [running Buildkit at scale](self-hosted/manage/buildkit-high-performance.mdx)  <!-- 8727 -->
+
+### Improvements
+
+- Okteto will now automatically create a Docker secret if one doesn't already exist in the Controller Manager. This prevents a misleading warning that was being displayed in some scenarios when deploying from the UI  <!-- 8566 -->
+- [Buildkit cache size is now automatically configured](self-hosted/helm-configuration.mdx#buildkit) based on its PVC volume size  <!-- 8723 -->
+- Removed Buildkit persistency as an installation step now that it defaults to true  <!-- 8771 -->
+- The Okteto frontend now runs rootless by default, while Buildkit operates without privileges when rootless mode is enabled  <!-- 8774, 8800 -->
+- We've [released Okteto CLI 3.1.0 with many new improvements](https://github.com/okteto/okteto/releases/tag/3.1.0)
+- [Okteto CLI 3.1.0](https://github.com/okteto/okteto/releases/tag/3.1.0): [Added support for a `context`](reference/okteto-manifest.mdx#context-string-optional) field in the Okteto Manifest, allowing you to specify the working directory for commands in the `deploy` and `destroy` sections
```
Comment (ifbyol) [inline src/content/release-notes.mdx:35]: The link of context points to `test` section. I think we are missing an update of the `deploy` and `destroy` command to add the `context` property, so I'd remove the link for now. I had added the link on my previous message just to illustrate that the one we added was similar to the one we already have for `test`

**Review by ifbyol — APPROVED** [review]

**Comment by ifbyol** [comment]
> @ifbyol I think the latest changes address your feedback. Please take a look and let me know what you think. Also, is there anything that we're not covering for CLI 3.1.0?

@pchico83 What is the plan for the override of the BuildKit frontend image? Should we document it or should just wait for next release for the docs about the air gapped scenarios?

File: src/content/release-notes.mdx [hunk 8]
```diff
@@ -5,6 +5,43 @@ sidebar_label: Release notes
...
+- Added Okteto [Garbage Collector settings to the Admin Dashboard](admin/cleanup.mdx): Admins can now manage sleep and delete periods from the Dashboard, with the ability to set different configurations for Namespaces and Preview Environments
+- [Remote Execution can now be made the default](core/remote-execution.mdx) by configuring it in the Okteto Admin Dashboard. This improves usability, particularly for private Git repositories and context synchronization with BuildKit, making remote deployments more consistent and reliable
```
Comment (pchico83) [inline src/content/release-notes.mdx:25]: I would rethink this paragraph. There are two things here:
- Having this option gives admin the power to force remote deploys everywhere. Remote deploys is our recommendation to have a more consistent deploy behavior across developers
- We have implemented two improvements on deploy on remote to provide feature parity with local deploys. They are `context` and the ability to close private git repositories as part of the deploy commands

**Comment by pchico83** [comment]
> > @ifbyol I think the latest changes address your feedback. Please take a look and let me know what you think. Also, is there anything that we're not covering for CLI 3.1.0?
> 
> @pchico83 What is the plan for the override of the BuildKit frontend image? Should we document it or should just wait for next release for the docs about the air gapped scenarios? That would be the only thing from the CLI not documented, but the most important things are already in the notes

@ifbyol it's documented as a feature flag for now. For the next release, we will have the full airgap guide

File: src/content/release-notes.mdx [hunk 9]
```diff
@@ -5,6 +5,43 @@ sidebar_label: Release notes
...
+## 1.26.0
+
+11 November 2024
+
+This versions is compatible with Kubernetes versions 1.27 to 1.30
+
+### Breaking Changes
+
+- **Hostname Length Limit**: Deploys now fail if a service hostname exceeds 63 characters, showing an error message. Previously, the dev environment deployed successfully even if the endpoint didn’t work. This may affect environments that deployed without issues before.
+- **Helm Release Name Limit**: Helm release names are now limited in length. Use defaultBackend.nameOverride to shorten the default backend service name if needed. If changing this during an upgrade, [follow the guide as it may impact the installation](self-hosted/helm-configuration.mdx#manual-migration-steps-when-renaming-the-defaultbackend-service).
```
Comment (andreafalzetti) [inline src/content/release-notes.mdx:17]: they can also use `fullnameOverride` or `nameOverride`, not necessarily have to rename the backend

File: src/content/release-notes.mdx [hunk 10]
```diff
@@ -5,6 +5,43 @@ sidebar_label: Release notes
...
+- **Buildkit Persistence Enabled**: Buildkit persistence is now enabled by default, with a 100Gi disk and cache set to 90% of the disk size. If you previously used buildkit.persistence.cache, adjust to the new ratio, as this setting is no longer applicable.
+
+### New Features
+
+- **[Introducing the Okteto Resource Manager](admin/resource-manager.mdx)**: a new feature that automatically optimizes CPU and memory requests for your environments. By analyzing real-time resource utilization, the Resource Manager dynamically adjusts resource requests to ensure efficient usage, prevent node overload, and improve overall cluster performance. This feature simplifies resource management, reduces manual adjustments, and enhances application stability, especially in larger clusters. The default installation provides recommendations but doesn't apply them automatically. [See our docs for details on how to apply these automatically](admin/resource-manager.mdx).
```
Comment (andreafalzetti) [inline src/content/release-notes.mdx:23]: # 🎉

File: src/content/release-notes.mdx (hunk 8 above)
Comment (codyjlandstrom) [reply src/content/release-notes.mdx:25]: Does this sound ok?
[Remote Execution can now be set as the default](core/remote-execution.mdx) in the Okteto Admin Dashboard, allowing admins to enforce consistent remote deploys. Improvements include feature parity with local deploys, such as context synchronization and private Git repository cloning during deploys

Comment (pchico83) [reply src/content/release-notes.mdx:25]: Looks good!

**Review by pchico83 — APPROVED** [review]

**Pull request merged**